extensions-core/kafka-extraction-namespace/src/main/java/io/druid/query/lookup/KafkaLookupExtractorFactory.java

```diff
@@ -194,7 +195,7 @@ public boolean start()
             public void run()
             {
               while (!executorService.isShutdown()) {
-                final ConsumerConnector consumerConnector = buildConnector(kafkaProperties);
+                consumerConnector = buildConnector(kafkaProperties);
```

Comment:
@drcrallen can you describe the race condition in more detail? from what I can see, the `synchronized(started)` along with `start()` not releasing the lock until after `startingReads` is cleared should prevent missing the connector opening.